internal/interpreter/api.rs

```diff
@@ -1035,6 +1035,9 @@ pub enum SetPropertyError {
     /// The property exist but does not have a type matching the dynamic value
     #[error("wrong type")]
     WrongType,
+    /// Attempt to set an output property
+    #[error("access denied")]
+    AccessDenied,
```

Comment:
I wonder if there should be different errors, one for attempted access to a private property and one for trying to write to an output property? The former could also produces “NoSuchProperty” instead.

Reply:
this is the case already.  Private == NoSuchProperty

Reply:
Ahh great